api/s2n.h

```diff
@@ -82,6 +82,9 @@ typedef enum { S2N_SERVER, S2N_CLIENT } s2n_mode;
 extern struct s2n_connection *s2n_connection_new(s2n_mode mode);
 extern int s2n_connection_set_config(struct s2n_connection *conn, struct s2n_config *config);
 
+extern int s2n_connection_set_ctx(struct s2n_connection *conn, void *ctx);
```

Comment:
One nit: a little blurb on what this does can be added to https://github.com/awslabs/s2n/blob/master/docs/USAGE-GUIDE.md